test/upgrade_test.go

```diff
@@ -0,0 +1,281 @@
+	filterPipelineRunStatusFields = cmpopts.IgnoreFields(v1.PipelineRunStatusFields{}, "StartTime", "CompletionTime", "Provenance")
+
+	simpleTaskYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  params:
+  - name: rsp
+    type: string
+    default: "response"
+  steps:
+  - name: echo-param
+    image: alpine
+    script: |
+      echo "$(params.rsp)"
+  - name: check-workspace
+    image: alpine
+    script: |
+      if [ "$(workspaces.taskWorkspace.bound)" == "true" ]; then
+        echo "Workspace provided"
+      fi
+  workspaces:
+  - name: taskWorkspace
+`
+
+	simplePipelineYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  tasks:
+  - name: %s
+    taskRef:
+      name: %s
+    workspaces:
+    - name: taskWorkspace
+      workspace: workspace
+  workspaces:
+    - name: workspace
+`
+
+	simpleTaskRunYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  taskRef:
+    name: %s
+    kind: Task
+  timeout: 30m
+  workspaces:
+    - name: taskWorkspace
+      emptyDir: {}
+`
+
+	simplePipelineRunYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  pipelineRef:
+    name: %s
+  workspaces:
+    - name: workspace
+      emptyDir: {}
+  timeouts:
+    pipeline: 180s
+`
+
+	expectedSimpleTaskRunYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  taskRef:
+    name: %s
+    kind: Task
+  timeout: 30m
+  workspaces:
+    - name: taskWorkspace
+      emptyDir: {}
+status:
+  conditions:
+  - type: Succeeded
+    reason: Succeeded
+    status: "True"
+  taskSpec:
+    steps:
+    - image: alpine
+      name: echo-param
+      script: |
+       echo "response"
+    - name: check-workspace
+      image: alpine
+      script: |
+        if [ "true" == "true" ]; then
+          echo "Workspace provided"
+        fi
+    workspaces:
+    - name: taskWorkspace
+    params:
+    - name: rsp
+      type: string
+      default: response
+  steps:
+  - container: step-echo
+    name: step-echo
+    terminated:
+      reason: Completed
+  - container: check-workspace
+    name: check-workspace
+    terminated:
+      reason: Completed
+`
+
+	expectedSimplePipelineRunYaml = `
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  timeouts:
+    pipeline: 3m
+  pipelineRef:
+    name: %s
+    kind: Pipeline
+  workspaces:
+    - name: workspace
+      emptyDir: {}
+status:
+  conditions:
+  - type: Succeeded
+    reason: Succeeded
+    status: "True"
+  pipelineSpec:
+    tasks:
+    - name: task1
+      taskRef:
+        name: task1
+        kind: Task
+      workspaces:
+      - name: taskWorkspace
+        workspace: workspace
+    workspaces:
+    - name: workspace
+      emptyDir: {}
+  childReferences:
+  - name: %s
+    pipelineTaskName: task1
+    apiVersion: "tekton.dev/v1"
+    kind: TaskRun
+`
+)
+
+// TestSimpleTaskRun creates a taskRun with a basic task and verifies the
+// runs created are successful and as expected.
+func TestSimpleTaskRun(t *testing.T) {
```

Comment:
I'm curious why running simple crds against previous version can fix https://github.com/tektoncd/pipeline/issues/4913, if there is a new non-nil field added in the new crd, and it is not in this test. Will it fail or pass?

Reply:
(sorry forgot to release my draft comment earlier)
The issue in #4913 was that the metadata(which is a common field to be used in the specifications) was not apparent with the newly introduced if condition in https://github.com/tektoncd/pipeline/pull/4834, so I think this is to prevent the regressions on basic specs and syntax. And we could build upon this simple test in the future. It has also been kept simple aiming not to introduce too much burden on the CI.